crates/proof_of_stake/src/lib.rs

```diff
@@ -303,6 +303,7 @@ where
         amount.change(),
         current_epoch,
```

Comment:
thinking about this, I think we want to check if the validator will be jailed between `current` and `proposal_end_epoch`